contracts/vaults/TreasuryVault/RibbonTreasuryVault.sol

```diff
@@ -339,13 +267,29 @@ contract RibbonTreasuryVault is
             "Invalid management fee"
         );
 
-        // We are dividing annualized management fee by num weeks in a year
-        uint256 tmpManagementFee =
-            newManagementFee.mul(Vault.FEE_MULTIPLIER).div(WEEKS_PER_YEAR);
+        managementFee = _perRoundManagementFee(newManagementFee);
 
         emit ManagementFeeSet(managementFee, newManagementFee);
+    }
 
-        managementFee = tmpManagementFee;
+    /**
+     * @notice Internal function to set the management fee for the vault
+     * @param managementFee is the management fee (6 decimals). ex: 2 * 10 ** 6 = 2
+     * @return perRoundManagementFee is the management divided by the number of rounds per year
+     */
+    function _perRoundManagementFee(uint256 managementFee)
+        internal
+        view
+        returns (uint256)
+    {
+        uint256 _period = period;
+        uint256 feeDivider =
+            _period % 30 == 0
+                ? Vault.FEE_MULTIPLIER.mul(12 / (_period / 30))
+                : WEEKS_PER_YEAR.div(_period / 7);
+
+        // We are dividing annualized management fee by num weeks in a year
+        return managementFee.mul(Vault.FEE_MULTIPLIER).div(feeDivider);
```

Comment:
We seem to be mixing between using `mul` `div` and normal signs. We should standardize them by doing `*` `/`, since the overflow checks are already done for us in Solidity 0.8